syntax = "proto3";
package rofi.messages;

import "jointSetPosWithSpeedCmd.proto";
import "jointSetVelocityCmd.proto";
import "jointSetTorqueCmd.proto";

message JointCmd {
    enum Type {
        NO_CMD = 0;
        GET_MAX_POSITION = 1;
        GET_MIN_POSITION = 2;
        GET_MAX_SPEED = 3;
        GET_MIN_SPEED = 4;
        GET_MAX_TORQUE = 5;
        GET_POSITION = 6;
        GET_VELOCITY = 7;
        GET_TORQUE = 8;
        SET_POS_WITH_SPEED = 9;
        SET_VELOCITY = 10;
        SET_TORQUE = 11;
    }

    int32 joint = 1;
    Type cmdType = 2;

    JointSetPosWithSpeedCmd setPosWithSpeed = 4;
    JointSetVelocityCmd setVelocity = 3;
    JointSetTorqueCmd setTorque = 5;
}

message PIDDebug {
    double positionError = 2;
    double velocityError = 3;
    double outputFromPosition = 4;
    double outputFromVelocity = 5;
    double outputFromForce = 6;
    double posPError = 7;
    double posIError = 8;
    double posDError = 9;
    double velPError = 10;
    double velIError = 11;
    double velDError = 12;
}
